File: recursion/standard-uses/binary_tree_traversal.py
from BST_classes import BST, Node # Uses a binary search tree

def traverse_in_order(node):
    ''' In-order binary tree traversal '''
    if node != None:
        traverse_in_order(node.left)
        print (node.data)
        traverse_in_order(node.right)


def test_data(bst):
    ''' Inserts some test data into tree '''
    bst.insert('Zac')
    bst.insert('Lucy')
    bst.insert('Anil')
    bst.insert('Boris')
 
       
if __name__ == '__main__':
    my_tree = BST()
    test_data(my_tree) # Create some test data
    start = my_tree.get_root() # Get start node
    traverse_in_order(start) 

File: recursion/standard-uses/binary_tree_traversal.cs
/*
Note: This file is designed to be copied out and compiled on your machine. 
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file. 

To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Compile the program
4. Run the program
*/

using System;

namespace Recursion
{
    class BinaryTreeTraversal
    {
        // The Main method is the entry point for all C# programs
        static void Main(string[] args) {
            BST myTree = new BST();
            TestData(myTree);  // Create some test data
            Node start = myTree.GetRoot();  // Get start node
            TraverseInOrder(start); 
        }
        

        // Inserts some data into the tree
        static void TestData(BST bst) {
            bst.Insert("Zac");
            bst.Insert("Lucy");
            bst.Insert("Anil");
            bst.Insert("Boris");
        }

        
        // In order binary search traversal
        static void TraverseInOrder(Node node) {    
            if (node != null) {
                TraverseInOrder(node.left);
                Console.WriteLine(node.data);
                TraverseInOrder(node.right);
            }
        }
        
        
    }
}
